opensrp-tasking/src/main/java/org/smartregister/tasking/util/IndicatorUtils.java

```diff
@@ -0,0 +1,180 @@
+
+        List<TaskDetails> taskDetailsList = new ArrayList<>();
+
+        for (Map.Entry<String, Set<Task>> entry : map.entrySet()) {
+
+            for (Task task : entry.getValue()) {
+
+                taskDetailsList.add(convertToTaskDetails(task));
+            }
+
+        }
+
+        return taskDetailsList;
+
+    }
+
+    /**
+     * Convert task to task details object
+     *
+     * @param task the task
+     * @return TaskDetails object
+     */
+
+    public static TaskDetails convertToTaskDetails(Task task) {
+
+        TaskDetails taskDetails = new TaskDetails(task.getIdentifier());
+
+        taskDetails.setTaskCode(task.getCode());
+        taskDetails.setTaskEntity(task.getForEntity());
+        taskDetails.setBusinessStatus(task.getBusinessStatus());
+        taskDetails.setTaskStatus(task.getStatus().name());
+        taskDetails.setStructureId(task.getStructureId());
+
+        return taskDetails;
+
+    }
+
+    public static IndicatorDetails processIndicators(List<TaskDetails> tasks) {
+
+        Map<String, TaskDetails> indicatorDetailsMap = new HashMap<>();
+
+        IndicatorDetails indicatorDetails = new IndicatorDetails();
+
+        if (tasks != null) {
+
+            for (int i = 0; i < tasks.size(); i++) {
+
+                if (Constants.Intervention.IRS.equals(tasks.get(i).getTaskCode())) {
+
+                    indicatorDetailsMap.put(tasks.get(i).getStructureId(), tasks.get(i));
+                }
+
+            }
+
+            for (Map.Entry<String, TaskDetails> entry : indicatorDetailsMap.entrySet()) {
+
+                if (Constants.BusinessStatusWrapper.SPRAYED.contains(entry.getValue().getBusinessStatus())) {
+
+                    indicatorDetails.setSprayed(indicatorDetails.getSprayed() + 1);
+
+                } else if (Constants.BusinessStatusWrapper.NOT_SPRAYED.contains(entry.getValue().getBusinessStatus())) {
+
+                    indicatorDetails.setNotSprayed(indicatorDetails.getNotSprayed() + 1);
+
+                } else if (Constants.BusinessStatusWrapper.NOT_VISITED.contains(entry.getValue().getBusinessStatus())) {
+
+                    indicatorDetails.setNotVisited(indicatorDetails.getNotVisited() + 1);
+
+                } else if (Constants.BusinessStatusWrapper.NOT_ELIGIBLE.contains(entry.getValue().getBusinessStatus())) {
+
+                    indicatorDetails.setIneligible(indicatorDetails.getIneligible() + 1);
+
+                }
+            }
+
+        }
+
+        indicatorDetails.setTotalStructures(indicatorDetailsMap.size() - indicatorDetails.getIneligible());
+        indicatorDetails.setProgress(indicatorDetails.getTotalStructures() > 0 ? Math.round(indicatorDetails.getSprayed() * 100 / indicatorDetails.getTotalStructures()) : 0);
+
+        return indicatorDetails;
+    }
+
+
+//    public static List<String> populateNamibiaSprayIndicators(Context context, IndicatorDetails indicatorDetails) {
+//        List<String> sprayIndicator = new ArrayList<>();
```

Comment:
I am not sure of the intention when commenting these out. Could you do something about it? Create an issue, todo, remove, fix